hephaestus/worker.py

```diff
@@ -68,7 +68,10 @@ def __init__(self, **kwargs):
 
     def run(self):
         while not _shutdownEvent.is_set():
-            message = self.messageQueue.get()
+            try:
+                message = self.messageQueue.get(True, 10)
+            except queue.Empty:
+                continue
```

Comment:
A debug log entry can be emitted here to signal that the MessageWorker is going to retry fetching from queue